### PR TITLE
adds includes container

### PR DESCRIPTION
**Pull request by rhenshaw56**
Resolves https://github.com/bbc/simorgh-infrastructure/issues/898

**Overall change:** 
_Adds container for includes._

**Code changes:**
- Created container for includes
- Added toggle for Includes
- Fetch include markup and add constraints that limits fetching markup when toggles are disabled

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] I have added labels to this PR for the relevant pod(s) affected by these changes
- [X] I have assigned this PR to the Simorgh project

**Testing:**
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/Include/index.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const logger = webLogger();
+
+const IncludeContainer = ({ href }) => {
+  const [markup, setMarkup] = useState(null);
+  const [hasError, setError] = useState(false);
+  const { enabled } = useToggle('include');
+
+  useEffect(() => {
+    const fetchMarkup = async () => {
```
Comment (andrewscfc) [inline src/app/containers/Include/index.jsx:17]: Did we ever have the discussion about whether to fetch the include clientside or serverside? I think @jamesdonoh wanted to dig into this subject and @hindsc52 had reservations about fetching serverside. Maybe we should arrange a call to go over this decision?

Right now am I correct in saying doing the `fetch` in a `useEffect` hook means the fetch will always occur clientside?

File: src/app/containers/Include/index.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        setError(true);
+        logger.error(`HTTP Error: "${e}"`);
+      }
+    };
+    if (enabled) {
+      fetchMarkup();
+    }
+  }, [href, enabled]);
+
+  const shouldNotDisplayInclude = hasError || !(markup && enabled);
```
Comment (andrewscfc) [inline src/app/containers/Include/index.jsx]:
```suggestion
  const shouldNotDisplayInclude = hasError || !markup || !enabled;
```
I struggled to follow this statement, my suggestion just represents:
Don't display the include if:
- We had an error fetching the include: `hasError`
- The markup is empty: `!markup`
- Includes not enabled: `!enabled`

File: src/app/containers/Include/index.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  if (shouldNotDisplayInclude) {
+    return null;
+  }
+
+  return (
+    <GridItemConstrainedMedium>
+      <div dangerouslySetInnerHTML={{ __html: markup }} />,
```
Comment (andrewscfc) [inline src/app/containers/Include/index.jsx:44]: So in this PR we expect the html to render with no scripts working? That was my understanding but just making sure 👍

File: src/app/containers/Include/index.test.jsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { render, wait } from '@testing-library/react';
+import IncludeContainer from '.';
+import { ToggleContext } from '#contexts/ToggleContext';
+import { RequestContextProvider } from '#contexts/RequestContext';
+
+const responseText = `<!DOCTYPE html><html lang="en"> <script type="text/javascript">var count=0; function incrementCount(cnt){count=cnt + 1; document.getElementById('count').innerHTML=count;}; function decrementCount(cnt){count=cnt - 1; document.getElementById('count').innerHTML=count;}; </script> <head> <style>html{display: inline-flex;}span{background-color: #333; border-radius: 3px; color: #fff; padding: 1px 5px;}button{background-color: #fff; border: 1px solid currentColor; border-radius: 3px; cursor: pointer; outline: 0;}button:active{background-color: #333; color: #fff;}button, span{margin: 0 2px;}</style> </head> <body> <div> <span id="count">0</span> <button onClick="incrementCount(count)">Increase Count</button> <button onClick="decrementCount(count)">Decrease Count</button> </div></body></html>`;
```
Comment (andrewscfc) [inline src/app/containers/Include/index.test.jsx]: Could we move this out to a seperate file so maintainers can see what the 'fake include' does, with it being minified people may assume it is a real include example? If you think this is a good idea, I wouldn't minify it so you can see what the fake include does.

File: src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js [hunk 5]
```diff
@@ -11,8 +18,8 @@ const convertInclude = ({ href, url, ...rest }) => {
   return {
     type,
     model: {
-      // `url` here should be replaced with `href` once mozart routes have been created. /*TODO: Create issue for this */
-      href: url,
+      // `url || includeUrls[type]` here should be replaced with `href` once mozart routes have been created. /*TODO: Create issue for this */
```
Comment (andrewscfc) [inline src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js:21]: Should an issue be created before this gets merged?

**Review by andrewscfc — COMMENTED** [review]
Few comments, nothing major though, looking good 🙂

File: src/app/containers/Include/index.jsx (hunk 3 above)
Comment (rhenshaw56) [reply src/app/containers/Include/index.jsx:44]: yes @andrewscfc that's also my understanding from the question I asked during standups

File: src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js (hunk 5 above)
Comment (rhenshaw56) [reply src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js:21]: yes, I will write up an issue for this to be refined and blocked pending the completion of the dependent issues.

Comment (rhenshaw56) [reply src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js:21]: created https://github.com/bbc/simorgh-infrastructure/issues/903 for this

Comment (andrewscfc) [reply src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js:21]: @rhenshaw56 don't think this is the right issue ^

Comment (rhenshaw56) [reply src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/index.js:21]: thanks, Andrew. It's been updated

**Pull request closed without merging**